kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -488,83 +494,102 @@ define([
+                };
+                if (config.initialDisplayValue) {
+                    if (typeof config.initialDisplayValue === 'string') {
+                        viewData.text = config.initialDisplayValue;
+                    } else if (
+                        config.initialDisplayValue !== null &&
+                        typeof config.initialDisplayValue === 'object'
+                    ) {
+                        viewData = Object.assign(viewData, config.initialDisplayValue);
+                        if (!viewData.text) {
+                            viewData.text = viewData.id;
                         }
                     }
-                    data.push(viewData);
                 }
-                dropdown.select2({
-                    allowClear: true,
-                    disabled: config.isViewOnly,
-                    templateResult: formatObjectDisplay,
-                    templateSelection: selectionTemplate,
-                    data: config.isViewOnly ? data : null,
-                    ajax: {
-                        delay: 250,
-                        processResults: (data) => {
-                            // update the currently selected one if applicable
-                            const currentData = dropdown.select2('data')[0];
-                            if (currentData) {
-                                const updatedData = data.results.filter((item) => {
-                                    return item.id === currentData.id;
-                                });
-                                if (updatedData.length) {
-                                    for (const key of Object.keys(updatedData[0])) {
-                                        currentData[key] = updatedData[0][key];
-                                    }
-                                    dropdown.trigger('change');
+                data.push(viewData);
+            } else if (model.value && !model.displayValue && ddOptions.exact_match_on) {
+                const searchResults = await fetchData(model.value);
+                // verify that we have an exact match
+                let match = null;
+                for (const result of searchResults) {
+                    if (
+                        model.value.toLowerCase() === result[ddOptions.exact_match_on].toLowerCase()
+                    ) {
+                        match = result;
+                        break;
+                    }
+                }
+                if (match) {
+                    data.push(Object.assign(match, { selected: true }));
+                    model.value = match.id;
+                } else {
+                    model.exactMatchError = true;
+                }
+            }
+
+            dropdown.select2({
+                allowClear: true,
+                disabled: config.isViewOnly,
+                templateResult: formatObjectDisplay,
+                templateSelection: selectionTemplate,
+                data,
```

Comment:
Turns out we can seed select2 with all the intro data we want, and the AJAX command will just updated it (with the `processResults` callback).